### PR TITLE
Add presence/absence matrix as input possibility

### PR DESCRIPTION
**Pull request by emilpaulitz**
Also add an examples and upgrade the tab example so it contains the same data as the roary mini example. This way, all of them actually find associations.

### CODE REVIEW

File: goldfinder/data_import.py [hunk 1]
```diff
@@ -21,10 +21,26 @@ def load_input(pinput, filetype, pmetadata, pknown_associations):
...
     elif filetype == "panx":
         df, locus_dict = load_panx(pinput)
+    elif filetype == "matrix":
+        df = pd.read_csv(pinput, index_col=0, header=0)
+        if df.index.has_duplicates:
+            exit("Error: input matrix contains non-unique gene names.")
+
+        """
+        # For future developers: Convert your matrix into the tab format with this code:
+        with open("C:/Users/emilp/Documents/Projekte/goldfinder_data/tab_example.tsv", 'w') as file:
```
Comment (fbaumdicker) [inline goldfinder/data_import.py]: I would remove your local path here and only keep the part within the goldfinder path

File: goldfinder/goldfinder.py [hunk 2]
```diff
@@ -23,9 +23,10 @@ def argparser():
     # Input
     parser.add_argument("-i", "--input", required=True,
                         help="File containing gene presence/absence data, or panX folder", default=argparse.SUPPRESS)
-    parser.add_argument("-f", "--file_type", nargs='?', choices=["roary", "tab", "panx"],
+    parser.add_argument("-f", "--file_type", nargs='?', choices=["roary", "tab", "panx", "matrix"],
                         default="roary", help="Clarifying which input file type Goldfinder is "
```
Comment (fbaumdicker) [inline goldfinder/goldfinder.py]: let's make matrix the default

File: README.md [hunk 3]
```diff
@@ -92,9 +92,10 @@ For more information on usage, please see `goldfinder -h`:
 File input:
   -i INPUT, --input INPUT
                         File containing gene presence/absence data, or panX folder
-  -f [{roary,tab,panx}], --file_type [{roary,tab,panx}]
+  -f [{roary,tab,panx,matrix}], --file_type [{roary,tab,panx,matrix}]
                         Clarifying which input file type Goldfinder is dealing with: a)
-                        roary (or panaroo), b) tab, c) panx. (default: roary)
+                        roary (or panaroo), b) tab, c) panx d) .csv matrix in pandas
+                        format. (default: roary)
```
Comment (fbaumdicker) [inline README.md]: switch to matrix as the default also here and also in the example call

**Comment by fbaumdicker** [comment]
thanks @emilpaulitz . I added a few smaller comments, otherwise this looks good.

File: goldfinder/data_import.py (hunk 1 above)
Comment (emilpaulitz) [reply goldfinder/data_import.py]: Good catch

**Comment by emilpaulitz** [comment]
Ready from my side

**Comment by fbaumdicker** [comment]
looks good to me. Can you rebase and possibly squash this @emilpaulitz before we merge this into main?

**Comment by emilpaulitz** [comment]
Done!

**Review by fbaumdicker — APPROVED** [review]

**Comment by fbaumdicker** [comment]
merged. :)

**Pull request merged**